src/qibolab/instruments/oscillator.py

```diff
@@ -1,7 +1,35 @@
 from dataclasses import dataclass
 from typing import Optional
 
-from qibolab.instruments.abstract import Instrument, InstrumentSettings
+from qibo.config import log
+
+from qibolab.instruments.abstract import (
+    Instrument,
+    InstrumentException,
+    InstrumentSettings,
+)
+
+
+class DummyDevice:
+    """Dummy device that does nothing but follows the QCoDeS interface."""
```

Comment:
Why do we need to emulate QCoDeS ourselves?

I would rather bridge our interface to QCoDeS' one, if we start adopting their drivers. But I don't see the advantage of replicating (for sure I'm missing something).

Reply:
Thanks for the review @AleCandido. The answer to this question may be relevant for a few of your other comments too.

Currently for the local oscillators we are using qcodes see https://github.com/qiboteam/qibolab/blob/8c6a7ca729eb0e922086dff215b5d40e5b1069b2/src/qibolab/instruments/rohde_schwarz.py#L8 and https://github.com/qiboteam/qibolab/blob/8c6a7ca729eb0e922086dff215b5d40e5b1069b2/src/qibolab/instruments/erasynth.py#L11
Our driver is just a wrapper above them to make them comply with the `Instrument` interface and allow instantiation before connecting to the actual instrument, in contrast to qcodes which requires connection in order to instantiate the object.

Here I am emulating qcodes because I wanted to lift all the driver methods to the (dummy) `LocalOscillator`. If you see the current main, ERASynth is almost a copy of R&S. To fix this, I lifted all the methods to `LocalOscillator` and only change the `self.device` attribute for the different models:
* `LO_SGS100A` from qcodes for R&S
* `ERASynthPlusPlus` from qcodes for ERASynth
* my `DummyDevice` for the dummy LO.

The `DummyDevice` needs to comply with the qcodes interface for this work. Strictly speaking I should inherit the `DummyDevice` from the abstract qcodes instrument `LO_SGS100A` and `ERASynthPlusPlus` inherit. This is moving towards #403.

Reply:
Ok, great!

I didn't realize the bridge was already in place, and I was definitely missing a piece. Thanks for the detailed explanation.

(I believe I was mostly confused by the usage of `DummyDevice` in `ERASynthEthernet`. I will reply about this in the related comment)

Reply:
Maybe even this one should inherit from the QCoDeS abstract `Instrument`. What do you think?

Reply:
One potential issue with this is that currently qcodes is not a dependency of qibolab. It is only an optional depedency for some instrument groups, but ideally the dummy platform/oscillator could work without it.

Also here we are using a minimal set of the original qcodes `Instrument`, which contains several other methods. So maybe we could leave as it is for now and check if it's worth inheriting qcodes objects later.